timemachine/fe/rbfe.py

```diff
@@ -743,27 +835,48 @@ def get_mean_final_barostat_volume_scale_factor(trajectories_by_state: Iterable[
         mean_final_barostat_volume_scale_factor = get_mean_final_barostat_volume_scale_factor(trajectories_by_state)
         assert (mean_final_barostat_volume_scale_factor is not None) == all(has_barostat_by_state)
 
-        # Use equilibrated samples and the average of the final barostat volume scale factors from bisection phase to
-        # initialize states for HREX
-        initial_states_hrex = [
-            replace(
-                initial_state,
+        def get_initial_state(lamb: float) -> InitialState:
+            state_idx = get_nearest_state_idx(lamb, initial_states)
+            nearest_state = initial_states[state_idx]
+            traj = trajectories_by_state[state_idx]
+            if np.isclose(nearest_state.lamb, lamb):
+                state = nearest_state
+            else:
+                # If the lambda value is different, reconstruct the initial state to the correct parameters
+                state = make_initial_state_fn(lamb, False)
+
+                # Verify that the forces of the nearest lambda value's frames are stable, since frames were not generated
+                # with the same parameters
+                du_dx, _ = state.to_bound_impl().execute(traj.frames[-1], traj.boxes[-1], compute_u=False)
```

Comment:
Had previously been concerned about taking the frame from the nearest initial state. In pratice (tested on hif2a) this seems to not be an issue.

Had talked to @maxentile offline about doing a probabilistic approach, but ended up being too slow to be practice.